modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuilder.java

```diff
@@ -98,10 +99,10 @@ void stop() {
     /**
      * Initializes the build of the index.
      */
-    void startIndexBuild(TableIndexView tableIndexView, TableImpl table) {
+    void startIndexBuild(TableIndexView tableIndexView, TableImpl table, IndexDescriptor indexDescriptor) {
```

Comment:
It looks like `tableIndexView` is only used to obtain index ID, which we can now take from `indexDescriptor`. I suggest to remove `tableIndexView` from parameters and pass index ID directly to the components that need it (like `BuildIndexTask`). This will allow us to reduce the usage of configuration even further.

Reply:
I think that in the near future this code will disappear, the building of indexes has changed a little in the https://github.com/apache/ignite-3/pull/2030

Reply:
Ok